tests/test_mmotifs.py

```diff
@@ -0,0 +1,137 @@
+import numpy as np
+import numpy.testing as npt
+import pytest
+
+from stumpy.mmotifs import mmotifs
+from stumpy.mstump import mstump
+from stumpy import match
+
+import naive
+
+
+# These are tests for multidimensional motif discovery
+
+
+def naive_match(Q, T, excl_zone, max_distance):
+    m = Q.shape[0]
+    D = naive.distance_profile(Q, T, m)
+
+    matches = []
+    for i in range(D.size):
+        dist = D[i]
+        if dist <= max_distance:
+            matches.append(i)
+
+    # Removes indices that are inside the exclusion zone of some occurrence with
+    # a smaller distance to the query
+    matches.sort(key=lambda x: D[x])
+    result = []
+    while len(matches) > 0:
+        o = matches[0]
+        result.append([D[o], o])
+        matches = [x for x in matches if x < o - excl_zone or x > o + excl_zone]
+
+    return np.array(result, dtype=object)
+
+
+test_data = [
+    (
+        np.array([0.0, 1.0, 0.0]),
+        np.array([0.0, 1.0, 0.0, -1.0, -1.0, 0.0, 1.0, 0.0, -0.5]),
+    ),
+    (
+        np.array([0.0, 1.0, 2.0]),
+        np.array([0.1, 1.0, 2.0, 3.0, -1.0, 0.1, 1.0, 2.0, -0.5]),
+    ),
+    (np.random.uniform(-1000, 1000, [8]), np.random.uniform(-1000, 1000, [64])),
+]
+
+
+def test_motifs_multidimensional_one_motif_all_dimensions():
+    # Find the two dimensional motif pair
+
+    # Arrange
+    motif_distances_expected = np.array([[0.0, 0.04540775]])
+    motif_indices_expected = np.array([[3, 10]])
+    motif_subspaces_expected = [np.array([1, 3])]
+    motif_mdls_expected = [np.array([176.0, 177.509775, 191.26466251, 235.01955001])]
+
+    # Act
+    T = np.array(
+        [
+            [5.0, 0.0, 3.0, 3.0, 7.0, 9.0, 3.0, 5.0, 2.0, 4.0, 7.0, 6.0, 8.0, 8.0, 1.0],
+            [
+                7.0,
+                3.0,
+                5.0,
+                9.0,
+                8.0,
+                7.0,
+                4.0,
+                5.0,
+                10.0,
+                8.0,
+                4.0,
+                3.0,
+                2.0,
+                0.0,
+                1.0,
+            ],
+            [6.0, 7.0, 7.0, 8.0, 1.0, 5.0, 9.0, 8.0, 9.0, 4.0, 3.0, 0.0, 3.0, 5.0, 0.0],
+            [
+                0.0,
+                1.0,
+                3.0,
+                2.0,
+                6.0,
+                1.0,
+                9.0,
+                10.0,
+                1.0,
+                2.0,
+                2.0,
+                5.0,
+                1.0,
+                0.0,
+                4.0,
+            ],
+        ]
+    )
+    m = 3
+
+    P, I = mstump(T, m)
+    motif_distances, motif_indices, motif_subspaces, motif_mdls = mmotifs(
+        T, P, I, max_distance=np.inf, max_matches=2, k=1
+    )
+
+    # Assert
```

Comment:
Yep, if the tests pass then you are on the right track!  The goal here is to compare your manual calculations (without calling the `mmotifs()` function) with what the `mmotifs()` function would produce.

Then, you try to test all possible combinations of inputs by hand (note your expected manual outputs) and then see if the function you are testing also produces the same expected outputs. If not then there is something wrong with your function. 

After that, the goal is to think about nasty edge cases and to write tests to ensure that we are handling those edge cases properly. We don't have to catch everything (it's impossible to) but at least think about how a user might use/abuse the function and hopefully we test and make sure that we consider all reasonable (i.e., rationally obvious) use cases and have good coverage on those. 

Also, as you may have noticed, the goal of having a "naive" implementation means that you don't have to do things manually and, instead, it allows you to try random inputs of different small arrays (in addition to a pre-determined array with known outputs). The only difference between the naive implementation vs your implementation should be speed (because the naive implementation would use a bunch of slow for-loops but is easier to read and follow versus faster algorithms are usually VERY HARD to read/follow since they often employ hyper specialized tricks).